pages/docs/index.md

```diff
@@ -526,11 +551,229 @@ Empty
 
 ---
 
+## Webhooks (Only Staging yet)
+When your app needs information about specific events that have occurred on a lab, it can subscribe to Trumelabs webhook topics as a mechanism for receiving near-real-time data about these events.
+
+### Headers
+```json
+{
+ "X-Signature": "",
+ "X-Topic-Name": "KIT_PROCESSED"
+}
+```
+**X-Signature**: The payload signed with the webhook signature.
+
+**X-Topic-Name**: The name of the topic.
+
+&nbsp;
+### Events
+**KIT_PROCESSED**: When a kit result is processed
+
+&nbsp;
+### Registering a webhook
+Register an endpoint to receive webhook event
+
+#### Request
+
+**URL**: `/v1/webhooks`
+**Method**: `POST`  
+**Authentication**: Required
+
+#### Parameters
+
+- **`endpoint`** (string): 
+  - **Description**: The endpoint url (Should be a POST endpoint)
+- **`event_name`** (string):
+	- **Description**: e.g: KIT_PROCESSED
+
+#### Example Request
+
+```bash
+curl --request POST \
+  --url 'http://localhost:8080/v1/webhooks?=' \
+  --header 'App-Key: <YOUR_API_KEY>' \
+  --header 'Content-Type: application/json' \
+  --header 'User-Agent: insomnia/10.0.0' \
+  --data '{
+	"endpoint": "https://<YOUR_HOST>/<YOUR_ENDPOINT>",
+	"event_name": "KIT_PROCESSED"
+}'
+```
+
+#### Response
+```json
+{
+	"signature": "ad8a7aa0f0288e45fa27407ee0983023"
+}
+```
+
+&nbsp;
+### Retrieve my webhooks
+Retrieve all registered webhooks 
+
+#### Request
+
+**URL**: `/v1/webhooks`
+**Method**: `GET`  
+**Authentication**: Required
+
+
+#### Example Request
+
+```bash
+curl --request GET \
+  --url 'http://localhost:8080/v1/webhooks' \
+  --header 'App-Key: 345' \
+  --header 'Content-Type: application/json' \
+  --header 'User-Agent: insomnia/10.0.0'
+```
+
+#### Response
+```json
+[
+	{
+		"endpoint": "http://localhost:8080/test-webhook",
+		"signature": "18fa91d5bc680a2c458e6763120d25ed",
+		"created_at": "2024-12-24T18:26:00"
+	}
+]
+```
+
+&nbsp;
+### Delete a webhook
+Delete a webhook
+
+#### Request
+
+**URL**: `/v1/webhooks`
+**Method**: `DELETE`  
+**Authentication**: Required
+
+
+#### Example Request
+
+```bash
+curl --request DELETE \
+  --url 'http://localhost:8080/v1/webhooks?=' \
+  --header 'App-Key: 345' \
+  --header 'Content-Type: application/json' \
+  --header 'User-Agent: insomnia/10.0.0' \
+  --data '{
+	"endpoint": "http://localhost:3000/test-webhook"
+}'
+```
+
+&nbsp;
+### Events Response
+**KIT_PROCESSED**:
+```json
+{
+	"bio_age_results": [
+		{
+			"kit_barcode": "TEST_WEBHOOK",
+			"status": "PROCESSED",
+			"chronological_age": 30.52,
+			"collection_date": "2024-12-23T21:51:45.608741",
+			"received_date": "2024-12-23T21:52:10.428050",
+			"biological_age": 30.0,
+			"processed_date": "2024-12-23T21:55:27.397521",
+			"peer_biological_age_score": 100.0,
+			"share_link": "http://localhost:3000/share/xtNvdxgF08ISJy"
```

Comment:
Maybe we should let users know that there is a possibility share_link won't come if user didn't register yet

Reply:
Adding a comment there.